### PR TITLE
:sparkles: Nightly ci initial structure

### PR DESCRIPTION
**Pull request by abrugaro**
First steps toward completing [Issue #8](https://github.com/konveyor/kai-ci/issues/8)

This PR introduces:

* Binary Downloading and Repository Cloning (part of #13):

    * Adds functionality to download binaries from a GitHub releases page
    
    * Implements repository cloning (for cloning the coolstore repository)
    
* CSV to JSON conversion and creation of report file for historical data (part of #11):

    * Converts an example csv report generated by kai-eval to json. This will be used to store historical data that will be used later for #15
    

* This PR Introduces a function to compress folders into a zip file. This will be used later to bundle the code and logs for upload to a cloud storage service

### CODE REVIEW

**Comment by abrugaro** [comment]
@sshveta @midays  could you take a look at this PR please? Thanks!

**Review by sshveta — APPROVED** [review]
LGTM

**Review by midays — APPROVED** [review]
Neat code! LGTM

File: kai-client-ci/main.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    test_app_name = os.getenv("APP_NAME")
+    test_original_branch = os.getenv("ORIGINAL_BRANCH")
+    test_repository_url = os.getenv("REPOSITORY_URL")
+
+    shutil.rmtree('data')
+    os.makedirs("data")
+
+    # download_binary()
+    clone_repository(test_app_name, test_repository_url, test_original_branch)
+
```
Comment (midays) [inline kai-client-ci/main.py:89]:
```suggestion
kai_eval_data = kai_eval_report_to_json()
total_incidents = len(kai_eval_data)

# Calculate average values
average_effectiveness = sum(item["Effectiveness"] for item in kai_eval_data) / total_incidents
average_specificity = sum(item["Specificity"] for item in kai_eval_data) / total_incidents
average_reasoning = sum(item["Reasoning"] for item in kai_eval_data) / total_incidents
average_competency = sum(item["Competency"] for item in kai_eval_data) / total_incidents
average_score = sum(item["Average Score"] for item in kai_eval_data) / total_incidents
```

File: kai-client-ci/main.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    json_report = {
+        "date": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+        "kaiEvalData": kai_eval_report_to_json()
```
Comment (midays) [inline kai-client-ci/main.py:92]:
```suggestion
    "Average Score": average_score,
    "TotalFiles": total_files,
    "TotalIncidents": total_incidents,
    "AverageEffectiveness": average_effectiveness,
    "AverageSpecificity": average_specificity,
    "AverageReasoning": average_reasoning,
    "AverageCompetency": average_competency,
    "kaiEvalData": kai_eval_data
```

**Review by midays — CHANGES_REQUESTED** [review]
Hi, I've added extra fields in the script to be added to the json

File: kai-client-ci/main.py (hunk 2 above)
Comment (abrugaro) [reply kai-client-ci/main.py:92]: Its ok for you if I implement these changes in the next PR?

**Review by midays — APPROVED** [review]

**Pull request merged**